src/components/Modals/FiatRamps/fiatRampProviders/banxa.ts

```diff
@@ -0,0 +1,50 @@
+import { adapters, caip19 } from '@shapeshiftoss/caip'
+import { ChainTypes } from '@shapeshiftoss/types'
+
+import { FiatRampAction, FiatRampAsset } from '../FiatRampsCommon'
+
+const banxaChainMap: Record<ChainTypes, string> = {
+  [ChainTypes.Ethereum]: 'ETH',
+  [ChainTypes.Bitcoin]: 'BTC',
+  [ChainTypes.Cosmos]: 'COSMOS',
+  [ChainTypes.Osmosis]: '',
+}
+
+export const getBanxaAssets = () => {
+  const banxaAssets = adapters.getSupportedBanxaAssets()
+  const assets: FiatRampAsset[] = banxaAssets.map(asset => ({
+    assetId: asset.CAIP19,
+    symbol: asset.ticker,
+    // name will be set in useFiatRampCurrencyList hook
+    name: '',
+  }))
+  return assets
+}
+
+export const createBanxaUrl = (action: FiatRampAction, asset: string, address: string): string => {
+  const BANXA_URL = 'https://shapeshift.banxa.com/'
+  let url = `${BANXA_URL}?`
+  url += `fiatType=USD&`
+  url += `coinType=${asset}&`
+  url += `walletAddress=${address}&`
```

Comment:
Not a fan of constructing a query string like this.
I get it why you used a let and conditionally constructing the qs, but there are better ways to do so. You used the `querystring` npm plugin in the gem refactor, I had a look at its docs and it should be able to make this cleaner